File: web/src/api/reports.ts
import { apiFetch, withEntityId } from './client'

export interface ReportGenerateRequest {
  entity_id?: string
  start_date: string
  end_date: string
}

export interface ReportResponse {
  id: string
  entity_id: string
  period_start: string
  period_end: string
  status: string
  filename: string | null
  generated_at: string | null
  error_message: string | null
}

export function generateReport(body: ReportGenerateRequest) {
  return apiFetch<ReportResponse>('/reports/generate', {
    method: 'POST',
    body: JSON.stringify(body),
  })
}

export function listReports(entityId?: string) {
  return apiFetch<ReportResponse[]>(withEntityId('/reports', entityId))
}

export function getReportStatus(id: string) {
  return apiFetch<ReportResponse>(`/reports/${id}/status`)
}

export function downloadReportUrl(id: string) {
  return `/api/reports/${id}/download`
}

File: src/cryptotax/api/schemas/reports.py
"""Schemas for /api/reports endpoints."""

import uuid
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ReportGenerateRequest(BaseModel):
    entity_id: Optional[uuid.UUID] = None
    start_date: str
    end_date: str


class ReportResponse(BaseModel):
    id: uuid.UUID
    entity_id: uuid.UUID
    period_start: datetime
    period_end: datetime
    status: str
    filename: Optional[str] = None
    generated_at: Optional[datetime] = None
    error_message: Optional[str] = None
